transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -230,11 +232,16 @@ private void doReleaseChannel(Channel channel, Promise<Void> promise) {
                          promise);
         } else {
             try {
-                if (offerChannel(channel)) {
-                    handler.channelReleased(channel);
-                    promise.setSuccess(null);
+                final Future<Boolean> f = healthCheck.isHealthy(channel);
```

Comment:
My vote is default should check the health. If this hurts performance in specific use cases it can be switched off. Our default health checker `ChannelHealthChecker.ACTIVE` shouldn't involve any network traffic so should be OK. @normanmaurer - WDYT?